src/c++/perf_analyzer/perf_utils.cc

```diff
@@ -200,6 +202,21 @@ SerializeExplicitTensor(
     std::copy(
         serialized.begin(), serialized.end(),
         std::back_inserter(*decoded_data));
+  } else if (dt.compare("JSON") == 0) {
+    std::string serialized = "";
+
+    for (const auto& value : tensor.GetArray()) {
+      rapidjson::StringBuffer buffer;
+      rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+      value.Accept(writer);
+
+      std::string element = buffer.GetString();
+      uint32_t len = element.size();
+      serialized.append(element);
```

Comment:
This just holds one string with a bunch of literal JSONs concatenated?

Can you walk me through where this ends up getting used and why it has to have the format that it does?

Reply:
Hard to show the entire path here, but this is the main part where it comes out:
https://github.com/triton-inference-server/client/blob/tgerdes-endpoint-scaffolding/src/c%2B%2B/perf_analyzer/client_backend/openai/openai_client_backend.cc#L67-L70

It ends up being the request body (since that is exactly the form that openai requests expect)

Reply:
But this looks like it will create one string with a bunch of concatenated request bodies, not just one request body. Either this shouldn't be a loop or I'm misunderstanding what serialized holds.

Reply:
I see your point. You are correct, I do not believe this would be valid JSON (or valid request body) if multiple json objects were in the input. I don't know what it looks like to fully support that in JSON or when it would be needed, so for now I will just add an assert to make sure there is only ever one entry.

Reply:
I have fixed it. I probably could clean up the format a bit so this wasn't necessary. I'll do that later in the week if I have time (it would require me changing some of Brian's code as well)